paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/ResourceFolderRepository.kt

```diff
@@ -75,4 +89,160 @@ class ResourceFolderRepository(
+    val fileNameToResourceName = SdkUtils.fileNameToResourceName(file.fileName)
+    return isIdentifier(fileNameToResourceName) && !isKeyword(fileNameToResourceName)
+  }
+
+  private class Loader(
+    private val repository: ResourceFolderRepository
+  ) : RepositoryLoader<ResourceFolderRepository>(
+    resourceDirectoryOrFile = repository.resourceDir.toPath(),
+    resourceFilesAndFolders = null,
+    namespace = repository.namespace
+  ) {
+    private val resourceDir: File = repository.resourceDir
+    private val resources = EnumMap<ResourceType, ListMultimap<String, ResourceItem>>(ResourceType::class.java)
+    private val sources = mutableMapOf<File, ResourceFile>()
+    private val fileResources = mutableMapOf<File, BasicFileResourceItem>()
+
+    // The following two fields are used as a cache of size one for quick conversion from a PathString to a File.
+    private var lastFile: File? = null
+    private var lastPathString: PathString? = null
+
+    fun load() {
+      if (!resourceDirectoryOrFile.exists()) {
+        return
+      }
+
+      scanResFolder()
+      populateRepository()
+    }
+
+    private fun scanResFolder() {
+      try {
+        for (subDir in resourceDir.listFiles()!!.sorted()) {
+          if (subDir.isDirectory) {
+            val folderName = subDir.name
+            val folderInfo = FolderInfo.create(folderName, folderConfigCache)
+            if (folderInfo != null) {
+              val configuration = getConfiguration(repository, folderInfo.configuration)
+              for (file in subDir.listFiles()!!.sorted()) {
+                if (file.name.startsWith(".")) {
+                  continue // Skip file with the name starting with a dot.
+                }
+                if (
+                  if (folderInfo.folderType == VALUES) {
+                    sources.containsKey(file)
+                  } else {
+                    fileResources.containsKey(file)
+                  }
+                ) {
+                  continue
+                }
+                val pathString = PathString(file)
+                lastFile = file
+                lastPathString = pathString
+                loadResourceFile(pathString, folderInfo, configuration)
+              }
+            }
+          }
+        }
+      } catch (e: Exception) {
+        LOG.severe("Failed to load resources from $resourceDirectoryOrFile: $e")
+      }
+
+      super.finishLoading(repository)
+
+      // Associate file resources with sources.
+      for ((file, item) in fileResources.entries) {
+        val source = sources.computeIfAbsent(file) { file ->
+          ResourceFile(file, item.repositoryConfiguration)
+        }
+        source.addItem(item)
+      }
+
+      // Populate the resources map.
+      val sortedSources = sources.values.toList()
+      // Sort sources according to folder configurations to have deterministic ordering of resource items in resources.
+      sortedSources.sortedWith(SOURCE_COMPARATOR)
```

Comment:
I think if you change this to `sortWith`, the `sorted` additions above can be reverted.  can you check?

Reply:
@jrodbx Could you check if the change to this part matches your thoughts? Thanks